### PR TITLE
Add route management for Android interface

### PR DESCRIPTION
**Pull request by pappz**
## Describe your changes

Support client route management feature on Android

### Checklist
- [ ] Is it a bug fix
- [ ] Is a typo/documentation fix
- [x] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: client/internal/routemanager/manager.go [hunk 1]
```diff
@@ -1,9 +1,130 @@
...
+	clientNetworks map[string]*clientNetwork
+	serverRouter   *serverRouter
+	statusRecorder *peer.Status
+	wgInterface    *iface.WGIface
+	pubKey         string
+}
+
+// NewManager returns a new route manager
+func NewManager(ctx context.Context, pubKey string, wgInterface *iface.WGIface, statusRecorder *peer.Status) *DefaultManager {
+	mCTX, cancel := context.WithCancel(ctx)
```
Comment (pascal-fischer) [inline client/internal/routemanager/manager.go:36]: I would use a more descriptive struct name than just Manager. Why not use RoutesManager like the package is called?

File: client/internal/routemanager/server_nonandroid.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		if err != nil {
+			return fmt.Errorf("couldn't initialize firewall containers, got err: %v", err)
+		}
+	}
+
+	for routeID := range m.routes {
+		update, found := routesMap[routeID]
+		if !found || !update.IsEqual(m.routes[routeID]) {
+			serverRoutesToRemove = append(serverRoutesToRemove, routeID)
+			continue
```
Comment (pascal-fischer) [inline client/internal/routemanager/server_nonandroid.go]: continue should have no effect

File: client/internal/routemanager/server_nonandroid.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+	for _, routeID := range serverRoutesToRemove {
+		oldRoute := m.routes[routeID]
+		err := m.removeFromServerNetwork(oldRoute)
+		if err != nil {
+			log.Errorf("unable to remove route id: %s, network %s, from server, got: %v",
+				oldRoute.ID, oldRoute.Network, err)
+		}
+		delete(m.routes, routeID)
+	}
```
Comment (pascal-fischer) [inline client/internal/routemanager/server_nonandroid.go:58]: Why do we first collect all toDeleteRoutes and then in a second run delete them? why not both in one loop? as we do for adding, here we do one round

Comment (pappz) [reply client/internal/routemanager/server_nonandroid.go:58]: This code come from the original implementation. The scope of this pull request is out of the code cleaning.

File: client/internal/routemanager/manager.go (hunk 1 above)
Comment (pappz) [reply client/internal/routemanager/manager.go:36]: This code come from the original implementation. The scope of this pull request is out of the code cleaning.

**Pull request merged**